app/util/Metadata.js

```diff
@@ -0,0 +1,265 @@
+            topic: 'gmd:identificationInfo/gmd:MD_DataIdentification/gmd:topicCategory/gmd:MD_TopicCategoryCode',
+            geography: {
+                extent: {
+                    minX: 'gmd:identificationInfo/gmd:MD_DataIdentification/gmd:extent/gmd:EX_Extent/gmd:geographicElement/gmd:EX_GeographicBoundingBox/gmd:westBoundLongitude/gco:Decimal',
+                    minY: 'gmd:identificationInfo/gmd:MD_DataIdentification/gmd:extent/gmd:EX_Extent/gmd:geographicElement/gmd:EX_GeographicBoundingBox/gmd:southBoundLatitude/gco:Decimal',
+                    maxX: 'gmd:identificationInfo/gmd:MD_DataIdentification/gmd:extent/gmd:EX_Extent/gmd:geographicElement/gmd:EX_GeographicBoundingBox/gmd:eastBoundLongitude/gco:Decimal',
+                    maxY: 'gmd:identificationInfo/gmd:MD_DataIdentification/gmd:extent/gmd:EX_Extent/gmd:geographicElement/gmd:EX_GeographicBoundingBox/gmd:northBoundLatitude/gco:Decimal'
+                },
+                projection: 'gmd:referenceSystemInfo/gmd:MD_ReferenceSystem/gmd:referenceSystemIdentifier/gmd:RS_Identifier/gmd:codeSpace/gco:CharacterString'
+            },
+            timeExtent: {
+                start: 'gmd:identificationInfo/gmd:MD_DataIdentification/gmd:extent/gmd:EX_Extent/gmd:temporalElement/gmd:EX_TemporalExtent/gmd:extent/gml:TimePeriod/gml:beginPosition',
+                end: 'gmd:identificationInfo/gmd:MD_DataIdentification/gmd:extent/gmd:EX_Extent/gmd:temporalElement/gmd:EX_TemporalExtent/gmd:extent/gml:TimePeriod/gml:endPosition'
+            },
+            format: 'gmd:distributionInfo/gmd:MD_Distribution/gmd:distributionFormat/gco:CharacterString',
+            limitations: 'gmd:metadataConstraints/gmd:MD_Constraints/gmd:useLimitation/gco:CharacterString',
+            onlineResource: 'gmd:dataSetURI:/gco:CharacterString',
+            dataSource: null,
+            publications: null
+        },
+
+        /**
+         *
+         */
+        parseXml: function(xmlString){
+            var parser = new DOMParser();
+            var xmlDoc = parser.parseFromString(xmlString, "text/xml");
+            return xmlDoc;
+        },
+
+        /**
+         *
+         */
+        sendCswRequest: function(xmlString, successCallBack){
+            Ext.Ajax.request({
+                url: BasiGX.util.Url.getWebProjectBaseUrl() +
+                    'metadata/csw.action',
+                method: "POST",
+                params: {
+                    xml: xmlString
+                },
+                defaultHeaders: BasiGX.util.CSRF.getHeader(),
+                scope: this,
+                success: successCallBack
+            });
+        },
+
+        /**
+         *
+         */
+        getInsertBlankXml: function(){
+            var xml;
+            Ext.Ajax.request({
+                url: BasiGX.util.Url.getWebProjectBaseUrl() + '/admin/resources/data/xmlTemplates/insertMetadata.xml',
+                method: 'GET',
+                async: false,
```

Comment:
It returns the UUID which is needed as a property for the layer to create. I tried to avoid nesting code, but this might break the layer creation. I'll fix this.